commons/src/main/java/org/codice/ddf/admin/commons/requests/RequestUtils.java

```diff
@@ -120,58 +127,38 @@ public ProbeReport sendGetRequest(String url, String userName, String password)
+        if (!c.isPresent()) {
+            return new ProbeReport().addMessage(createRequestConfigMsg(CANNOT_CONNECT));
         }
-        try {
-            post.setEntity(new StringEntity(content));
-        } catch (UnsupportedEncodingException e) {
-            return new ProbeReport(ConfigurationMessage.createInternalErrorMsg("Error encoding query."));
+        WebClient client = c.get();
+
+        Response r = client.type(contentType).post(content);
+        if (r.getStatus() == HTTP_OK) {
+            return new ProbeReport().addMessage(createRequestConfigMsg(EXECUTED_REQUEST))
+                    .probeResults(responseToMap(r));
         }
-        return sendHttpRequest(post);
+        return new ProbeReport().addMessage(createRequestConfigMsg(CANNOT_CONNECT));
     }
 
-    protected ProbeReport sendHttpRequest(HttpRequestBase request) {
-        ProbeReport results = new ProbeReport();
-        CloseableHttpClient client = null;
-        CloseableHttpResponse response = null;
+    private Optional<WebClient> generateClient(String url, String username, String password, SecurityManager securityManager) {
+        clientFactory = (username == null && password == null) ?
+                new SecureCxfClientFactory<>(url, WebClient.class) :
+                new SecureCxfClientFactory<>(url, WebClient.class, username, password);
+
         try {
-            client = getHttpClient(false);
-            response = client.execute(request);
-            if (response.getStatusLine().getStatusCode() == HTTP_OK) {
-                return new ProbeReport().addMessage(createRequestConfigMsg(EXECUTED_REQUEST))
-                        .probeResults(responseToMap(response));
-            } else {
-                return new ProbeReport().addMessage(createRequestConfigMsg(CANNOT_CONNECT));
-            }
-        } catch (SSLPeerUnverifiedException e) {
-            return results.addMessage(createRequestConfigMsg(CERT_ERROR));
-        } catch (IOException e) {
-            closeClientAndResponse(client, response);
-            try {
-                client = getHttpClient(true);
-                response = client.execute(request);
-                if (response.getStatusLine().getStatusCode() == HTTP_OK) {
-                    results.addMessage(createRequestConfigMsg(UNTRUSTED_CA));
-                    results.addMessage(createRequestConfigMsg(EXECUTED_REQUEST));
-                    return results.probeResults(responseToMap(response));
-                } else {
-                    return results.addMessage(createRequestConfigMsg(CANNOT_CONNECT));
-                }
-            } catch (Exception e1) {
-                return results.addMessage(createRequestConfigMsg(CANNOT_CONNECT));
-            }
-        } finally {
-            closeClientAndResponse(client, response);
+            Subject guestSubject = securityManager.getSubject(new GuestAuthenticationToken(
+                    BaseAuthenticationToken.ALL_REALM, "127.0.0.1"));
```

Comment:
If I'm not mistaken, this will generate a local guest auth token. I imagine this would allow you to connect to a non-local host that allowed guest authentication but I don't know for sure. This would be a good question for @stustison